TypeCobol/Compiler/CodeModel/SymbolTable.cs

```diff
@@ -621,72 +613,61 @@ private void AddAllReference(IList<DataDefinition> found, DataDefinition heaData
         /// <summary>Get all items with a specific name that are subordinates of a custom type</summary>
         /// <param name="name">Name of items we search for</param>
         /// <returns>Direct or indirect subordinates of a custom type</returns>
-        private List<DataDefinition> GetCustomTypesSubordinatesNamed(string name)
+        private List<DataDefinition> GetCustomTypesSubordinatesNamed(QualifiedName name)
         {
-            var subs = new List<DataDefinition>();
+            var foundDataDef = new List<DataDefinition>();
 
             //Get programs from Namespace table
             var programList = this.GetProgramsTable(GetTableFromScope(Scope.Namespace));
             foreach (var programs in programList) {
 
                 //Get Custom Types from program 
                 foreach (var pgm in programs.Value) { //we shouldn't have more than one program with the same name
-                                                      //but just in case it changes 
-                    seekSymbolTable(pgm.SymbolTable, name, subs);
+                    //but just in case it changes 
+                    foundDataDef.AddRange(SeekSymbolTable(pgm.SymbolTable, name));
                 }
             }
 
-
+            
             //Get Custom Types from Intrinsic table 
             var intrinsicTable = this.GetTableFromScope(Scope.Intrinsic);
-            foreach (var type in intrinsicTable.Types)
-            {
-                foreach (var type2 in type.Value)
-                {
-                    subs.AddRange(type2.GetChildren<DataDefinition>(name, true));
-                }
-            }
-
-            return subs;
+            foundDataDef.AddRange(GetCustomTypesFromSymbolTable(name, intrinsicTable));
+            return foundDataDef;
+        }
+        
+        /// <summary>
+        /// Search the qualified name in the DataTypeEntries dictionary
+        /// </summary>
+        /// <param name="name">QualifiedName</param>
+        /// <param name="table">SymbolTable that can have the searched DataDefinition</param>
+        /// <returns>List of found DataDefinitions that correspond to the passed name</returns>
+        private static IEnumerable<DataDefinition> GetCustomTypesFromSymbolTable(QualifiedName name, SymbolTable table)
+        {
+            //match only last part of the qualified name - the most specific one regardless of the case
+            return table.DataTypeEntries.Where(elem => elem.Key.ToArray().Last().Equals(name.ToArray().Last(),StringComparison.OrdinalIgnoreCase)).SelectMany(def => def.Value);
```

Comment:
As the key and the parameter of this method would be string (see previous remarks), then you use TryGet instead of a Where.